Package.swift

```diff
@@ -1,6 +1,8 @@
-// Package.swift - description
+// swift-tools-version:4.2
```

Comment:
I find the combinations of these version mechanisms so difficult to reason about. So this seems OK, but can you answer a question? What is the expectation when someone is using Swift tools, say, 4.1.2? If such an spm sees this package declaration at a given version tag, it will decide that it cannot use it, and search versions in reverse until it finds presumably the old Package.swift? Which will be at 1.2.0 (in a typical case where no other package added weird constraints)?

I mean, I think that is what the documentation says will happen I just want to see what you think. And what your intention is.

Also, if I am right, does that mean that spms version 4.0 through 4.1.2 are essentially pinned to SwiftProtobuf 1.2.0, but that spms in v3 are not, due to Package@swift-3.swift?

Or am I just wrong about some mechanism here?

Reply:
The only docs I can really find about it are https://github.com/apple/swift-package-manager/blob/master/Documentation/Usage.md#handling-version-specific-logic.

As far as your final question, I'm not completely sure what you are asking.  Since SwiftPM also looks for tags with this naming, we always have the option to cut a tag specific to a version and land a `Package` file on there for them if we need it.

As far as spms 4.0 - 4.1.2, since we have a `Package@swift-4.swift`, that will still work for them until we remove it.  And according to our version policy, that won't happen until we get a  5.2 or higher.

Reply:
No worries.  The @Swift-4.2 and non qualified are a little redundant at the moment, but I didn't see a way around that given we're trying to match older and newer behaviors.  :(